File: leetcode/array/1198_Find_Smallest_Common_Element_in_All_Rows/codes/arr.py
from collections import Counter
from typing import List

class Solution:
    def smallestCommonElement(self, mat: List[List[int]]) -> int:
        count = Counter()
        m = len(mat)
        
        for row in mat:
            for num in row:
                count[num] += 1
        
        for num in sorted(count):
            if count[num] == m:
                return num
        return -1

File: leetcode/array/1198_Find_Smallest_Common_Element_in_All_Rows/codes/arr.c
int smallestCommonElement(int** mat, int matSize, int* matColSize) {
    int count[10001] = {0};

    for (int i = 0; i < matSize; i++) {
        for (int j = 0; j < matColSize[i]; j++) {
            count[mat[i][j]]++;
        }
    }

    for (int i = 1; i <= 10000; i++) {
        if (count[i] == matSize) {
            return i;
        }
    }

    return -1;
}
